From f55c472258b097d41a3c3ceda2e35400a9aa79f2 Mon Sep 17 00:00:00 2001
From: shengfei Xu <xsf@rock-chips.com>
Date: Mon, 10 Apr 2017 11:12:48 +0800
Subject: [PATCH] ARM: rockchip: pm: add system suspend support for rk3288

PSCI v1.0 introduces a new API called PSCI_SYSTEM_SUSPEND. This API
provides the mechanism by which the calling OS can request entry into
the deepest possible system sleep state.

Change-Id: I2dbb56ad337315eee76170443de96a1df05f8aab
Signed-off-by: shengfei Xu <xsf@rock-chips.com>
---
 arch/arm/mach-rockchip/pm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-rockchip/pm.c b/arch/arm/mach-rockchip/pm.c
index be28a78f50b0..8d956f7a13e9 100644
--- a/arch/arm/mach-rockchip/pm.c
+++ b/arch/arm/mach-rockchip/pm.c
@@ -27,6 +27,7 @@
 #include <asm/cacheflush.h>
 #include <asm/tlbflush.h>
 #include <asm/suspend.h>
+#include <asm/psci.h>
 
 #include "pm.h"
 #include "embedded/rk3288_resume.h"
@@ -348,6 +349,9 @@ void __init rockchip_suspend_init(void)
 	struct device_node *np;
 	int ret;
 
+	if (psci_smp_available())
+		return;
+
 	np = of_find_matching_node_and_match(NULL, rockchip_pmu_of_device_ids,
 					     &match);
 	if (!match) {
-- 
2.35.3

